jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3461,9 +3461,16 @@ private static MethodHandle validateArgumentsOfWhileLoop(MethodHandle predHandle
 		Class<?> bodyReturnType = bodyType.returnType;
 		
 		/* The signature of the body handle must be either (V, A...)V or (A...)void */
-		if ((void.class != bodyReturnType) && (bodyReturnType != bodyType.arguments[0])) {
+		if ((void.class != bodyReturnType) 
+				&& (hasNoArgs(bodyType)
+						|| (bodyReturnType != bodyType.arguments[0]))) {
+			String argList = "void";
```

Comment:
In the case of void.class != bodyReturnType,   hasNoArgs(bodyType) and (bodyReturnType != bodyType.arguments[0]) should be considered as two different error cases.  Please split with K065D1 and K065D2.